orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -502,8 +504,9 @@ IMPLEMENT_GRADIENT_BUILDER(GetConcatTrainingGradient) {
   for (int i = 0; i < GetSrcNodeInputSize(); ++i) {
     std::vector<Dimension> data_shape;
     if (GetShape(I(i), data_shape).IsOK()) {
-      int64_t axis_index = axis < 0 ? static_cast<int64_t>(data_shape.size()) + axis : axis;
-      if (axis_index >= 0 && axis_index < static_cast<int64_t>(data_shape.size()) && data_shape[axis_index].has_dim_value()) {
+      int64_t rank = static_cast<int64_t>(data_shape.size());
+      int64_t axis_index = HandleNegativeAxis(axis, rank);
+      if (axis_index >= 0 && axis_index < rank && data_shape[axis_index].has_dim_value()) {
```

Comment:
`axis_index >= 0 && axis_index < rank` should already be ensured by HandleNegativeAxis()